src/fmeval/eval_algorithms/common.py

```diff
@@ -98,12 +99,16 @@ def evaluate_dataset(
         Currently, only MEAN is supported.
     :param save: If set to true, prompt responses and scores will be saved to a file.
         The path that this file is stored at is configured by `eval_results_path`.
+    :param col_to_validate: The column to validate for model inference. This can be the model input if
+        the model is provided, the model output if the model is not provided, or None if the model output is not
+        required by the evaluation algorithm.
 
     :return: An EvalOutput object encapsulating the results of the evaluation.
     """
     if model:
         try:
-            validate_dataset(dataset, [DatasetColumns.MODEL_INPUT.value.name])
+            col_to_validate = DatasetColumns.MODEL_INPUT.value.name
```

Comment:
While overriding `col_to_validate` here works, it is rather confusing in my opinion. I think it will be cleaner if we have two separate column-related parameters, `model_input_col` and `model_output_col`. Call `validate_dataset(dataset, [model_input_col])` when a model is present and `validate_dataset(dataset, [model_output_col])` otherwise.